### ISSUE TITLE
Allow `localhost` as request origin in the API headers

### ISSUE THREAD

**Issue by wholesomedev** [hollowverse-archive/hollowverse#234]
Should we allow `localhost` as a request origin in the API headers so that `yarn dev` wouldn't require a flaky browser extension to enable development?

**Comment by forabi** [comment]
Even if we add `localhost` to the list of allowed origins, [it won't work in Chrome](https://stackoverflow.com/questions/10883211/deadly-cors-when-http-localhost-is-the-origin):

> Chrome does not support localhost for CORS requests (an open bug since 2010).
>
> To get around this you can use a domain like lvh.me (which points at 127.0.0.1 just like localhost) or start chrome with the --disable-web-security flag (assuming you're just testing).

The mentioned bug is marked as "WontFix".

**Comment by wholesomedev** [comment]
There seems to be mixed reports around whether Chrome works correctly with `localhost` or not. It seems that the source of confusion is that people usually have a port number along with `localhost` and some people are unaware that a different port number constitutes a different origin. If we can handle that in our

```
Access-Control-Allow-Origin: http://localhost:3001
```

...things might work?

**Comment by forabi** [comment]
I never thought about that. I'll give it a try.

**Comment by wholesomedev** [comment]
Or when the API sees `localhost`, just respond with 

```
Access-Control-Allow-Origin: *
```

**Comment by forabi** [comment]
I've been looking into this, and I think a better solution would be to use the [Webpack DevServer `proxy` option](https://webpack.js.org/configuration/dev-server/#devserver-proxy).

We can configure the development build to use a path on the same origin as the web app, like `/__api/`, instead of `api.hollowverse.com/graphql`, and the development server would proxy all requests to `/__api/` to the production API. This way the API and the web app would always be on the same origin.

Why is this better than allowing `localhost`?

Allowing `localhost` only fixes the CORS issue when the dev server is running on `localhost`. During development, and for purposes of testing on real devices, I had to access builds running on my laptop from iPhone and Android devices. I was able to access the server using the local dev server IP, which in my case was `192.162.1.103`.

Since this IP is a different host from `localhost`, the requests to the API would fail even if we were to allow `localhost`, and using a browser extension is not an option on mobile devices either, so the only way to fix this is to use a proxy server.
